tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,297 @@
+  const auto nn_dist = make_not_null(&dist);
+  const auto nn_dist_pert = make_not_null(&dist_pert);
+
+  const auto box = db::create<
+      db::AddSimpleTags<StrahlkorperTags::items_tags<Frame::Inertial>>,
+      db::AddComputeTags<
+          StrahlkorperTags::compute_items_tags<Frame::Inertial>>>(strahlkorper);
+  const auto& unnormalized_normal =
+      db::get<StrahlkorperTags::NormalOneForm<Frame::Inertial>>(box);
+
+  const auto one =
+      make_with_value<Scalar<DataVector>>(unnormalized_normal, 1.0);
+  const auto minus_one =
+      make_with_value<Scalar<DataVector>>(unnormalized_normal, -1.0);
+  const auto five =
+      make_with_value<Scalar<DataVector>>(unnormalized_normal, 5.0);
+  const auto minus_five =
+      make_with_value<Scalar<DataVector>>(unnormalized_normal, -5.0);
+
+  // Choose spacetime_metric randomly, but make sure the result is
+  // still invertible. To do this, start with
+  // Minkowski, and then add a 10% random perturbation.
+  auto spacetime_metric_int = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+  get<0, 0>(spacetime_metric_int) += get(minus_one);
+  get<1, 1>(spacetime_metric_int) += get(one);
+  get<2, 2>(spacetime_metric_int) += get(one);
+  get<3, 3>(spacetime_metric_int) += get(one);
+
+  auto spacetime_metric_ext = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+  get<0, 0>(spacetime_metric_ext) += get(minus_one);
+  get<1, 1>(spacetime_metric_ext) += get(one);
+  get<2, 2>(spacetime_metric_ext) += get(one);
+  get<3, 3>(spacetime_metric_ext) += get(one);
+
+  // Set pi, phi to be random (phi, pi should not need to be consistent with
+  // spacetime_metric for the flux consistency tests to pass)
+  const auto phi_int = make_with_random_values<
+      tnsr::iaa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+  const auto pi_int = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+  const auto phi_ext = make_with_random_values<
+      tnsr::iaa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+  const auto pi_ext = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, unnormalized_normal);
+
+  const auto& spatial_metric_int = gr::spatial_metric(spacetime_metric_int);
+  const auto& inverse_spatial_metric_int =
+      determinant_and_inverse(spatial_metric_int).second;
+  const DataVector one_over_one_form_magnitude_int =
+      get(magnitude(unnormalized_normal, inverse_spatial_metric_int));
+  const auto unit_normal_one_form_int = StrahlkorperGr::unit_normal_one_form(
+      unnormalized_normal, one_over_one_form_magnitude_int);
```

Comment:
That might be because I'm showing that normal dot numerical flux reduces to normal dot flux, so an overall normalization in the normal applies to both. Is there a specific change you would like in response to this comment?

Reply:
I suggest to produce random `unit_one_form`s (that perhaps don't even need to be normalized) instead of constructing the `Strahlkorper`. Then the only thing this test needs as input is a number of grid points (or a `DataVector used_for_size`).

Reply:
I did that before seeing your comment :)